web/src/settings_banner.ts

```diff
@@ -0,0 +1,85 @@
+import $ from "jquery";
+
+import * as banners from "./banners.ts";
+import type {Banner} from "./banners.ts";
+import {$t} from "./i18n.ts";
+import * as settings_config from "./settings_config.ts";
+import * as settings_data from "./settings_data.ts";
+import {realm} from "./state_data.ts";
+
+const UPGRADE_ACCESS_BANNER: Banner = {
+    intent: "info",
+    label: $t({defaultMessage: "Available on Zulip Cloud Standard."}),
+    buttons: [
+        {
+            label: $t({defaultMessage: "Upgrade to access"}),
+            custom_classes: "request-upgrade",
+            attention: "quiet",
+        },
+    ],
+    custom_classes: "organization-upgrade-banner",
+    close_button: false,
+};
+
+const AVAILABLE_ON_STANDARD: Banner = {
+    intent: "neutral",
+    label: $t({defaultMessage: "Available on Zulip Cloud Standard."}),
+    buttons: [],
+    custom_classes: "organization-upgrade-banner",
+    close_button: false,
+};
+
+const UPGRADE_OR_SPONSORSHIP_BANNER: Banner = {
+    intent: "info",
+    label: $t({
+        defaultMessage: "Available on Zulip Cloud Standard.",
+    }),
+    buttons: [
+        {
+            label: $t({defaultMessage: "Upgrade"}),
+            custom_classes: "request-upgrade",
+            attention: "quiet",
+        },
+        {
+            label: $t({defaultMessage: "Request sponsorship"}),
+            custom_classes: "request-sponsorship",
+            attention: "borderless",
+        },
+    ],
+    custom_classes: "organization-upgrade-banner",
```

Comment:
I feel like this code could be refactored; it'd be a lot cleaner to just define `upgrade_buttons`, the only part that varies, based on the parameters, without repeating the rest of this definition.